src/sdk/analytics/platform/vtex/search.ts

```diff
@@ -0,0 +1,70 @@
+/**
+ * More info at: https://www.notion.so/vtexhandbook/Event-API-Documentation-48eee26730cf4d7f80f8fd7262231f84
+ */
+
+import config from 'src/../store.config'
+
+const THIRTY_MINUTES_S = 30 * 60
+const ONE_YEAR_S = 365 * 24 * 3600
+
+const randomUUID = () =>
+  typeof crypto.randomUUID === 'function'
+    ? crypto.randomUUID()
+    : (Math.random() * 1e6).toFixed(0)
+
+const storage = (key: string, expiresSecond: number) => {
+  const timelapsed = (past: number) => (Date.now() - past) / 1e3
+
+  return () => {
+    const item = JSON.parse(localStorage.getItem(key) ?? 'null')
+    const isExpired = !item || timelapsed(item.createdAt) > expiresSecond
+    const payload: string = isExpired ? randomUUID() : item.payload
+
+    if (isExpired) {
+      const data = { payload, createdAt: Date.now() }
+
+      localStorage.setItem(key, JSON.stringify(data))
+    }
+
+    return payload
+  }
+}
+
+const user = {
+  anonymous: storage('vtex.search.anonymous', ONE_YEAR_S),
+  session: storage('vtex.search.session', THIRTY_MINUTES_S),
+}
+
+type SearchEvent =
+  | {
+      type: 'session.ping'
+    }
+  | {
+      position: number
+      productId: string
+      text: string
+      url: string
+      type: 'search.click'
+    }
+  | {
+      text: string
+      match: number
+      misspelled?: boolean
+      operator: 'and' | 'or'
+      url: string
+      type: 'search.query'
+    }
+
+export const send = (options: SearchEvent & { url?: string }) =>
```

Comment:
How about add this function to the party town?

Reply:
How can we do it?

Reply:
The party town proxy functions inside the window. So can transform it as a global function, or transform it in a script inside JS file and fetch it through  tag `script` .

Reply:
If we do this, we will need a script tag and move this `send` function into this script, right? I'm not sure if we need to do this since this function is hard-coded by us and pretty small and lazy-loaded

Reply:
Ok, no problem, my main concern is about the functions to create random UUID and other small stuff. But that's ok